Frontend/ui-customer/src/Screens/RentalBookingFlow/RentalScreen/Controller.purs

```diff
@@ -0,0 +1,104 @@
+import Components.PrimaryButton.Controller as PrimaryButtonController
+import Data.Eq.Generic (genericEq)
+import Data.Generic.Rep (class Generic)
+import Data.Maybe (Maybe(..))
+import Data.Show.Generic (genericShow)
+import Debug (spy)
+import Effect.Aff (launchAff)
+import Log (trackAppActionClick, trackAppEndScreen)
+import Prelude (class Eq, class Show, class Show, bind, map, pure, unit, (<), (>), (+), (-), ($))
+import PrestoDOM (class Loggable, Eval, continue, continueWithCmd, exit)
+import PrestoDOM.Core (getPushFn)
+import Screens (getScreen, ScreenName(..))
+import Screens.Types (RentalScreenStage(..), RentalScreenState)
+
+instance showAction :: Show Action where
+  show _ = ""
+
+instance loggableAction :: Loggable Action where
+  performLog action appId = case action of
+    GenericHeaderAC _ -> trackAppActionClick appId (getScreen RIDE_SCHEDULED_SCREEN) "BackClicked" "on_click"
+    DurationIncrementDecrementAC _ -> trackAppActionClick appId (getScreen RIDE_SCHEDULED_SCREEN) "DurationIncrementDecrement" "on_click"
+    DistanceIncrementDecrementAC _ -> trackAppActionClick appId (getScreen RIDE_SCHEDULED_SCREEN) "DistanceIncrementDecrement" "on_click"
+    ChooseVehicleAC _ -> trackAppActionClick appId (getScreen RIDE_SCHEDULED_SCREEN) "ChooseVehicle" "on_click"
+    DateTimePickerAction _ _ _ _ _ _ _ -> trackAppActionClick appId (getScreen RIDE_SCHEDULED_SCREEN) "DateTimePicker" "on_click"
+    _ -> pure unit
+
+data Action = 
+    NoAction 
+  | PrimaryButtonActionController PrimaryButtonController.Action
+  | GenericHeaderAC GenericHeaderController.Action
+  | DurationIncrementDecrementAC IncrementDecrementModelController.Action
+  | DistanceIncrementDecrementAC IncrementDecrementModelController.Action
+  | ChooseVehicleAC ChooseVehicleController.Action
+  | DateTimePickerAction String Int Int Int String Int Int
+
+data ScreenOutput = NoScreen
+
+data FareBreakupRowType = BookingTime | BookingDistance | BaseFare | TollFee
+
+derive instance genericFareBreakupRowType :: Generic FareBreakupRowType _
+instance showFareBreakupRowType :: Show FareBreakupRowType where show = genericShow
+instance eqFareBreakupRowType :: Eq FareBreakupRowType where eq = genericEq
+
+eval :: Action -> RentalScreenState -> Eval Action ScreenOutput RentalScreenState
+
+eval (PrimaryButtonActionController (PrimaryButtonController.OnClick)) state = 
+  case state.data.currentStage of
+    RENTAL_SELECT_PACKAGE -> continue state { data { currentStage = RENTAL_SELECT_VARIANT }}
+    RENTAL_SELECT_VARIANT -> continue state { data { currentStage = RENTAL_CONFIRMATION }}
+    RENTAL_CONFIRMATION -> exit NoScreen
+
+eval (DurationIncrementDecrementAC (IncrementDecrementModelController.OnIncrement)) state =
+  let updatedDuration = if(state.data.rentalBookingData.baseDuration < state.props.maxDuration) then state.data.rentalBookingData.baseDuration + 1 else state.data.rentalBookingData.baseDuration
```

Comment:
`(state.data.rentalBookingData.baseDuration < state.props.maxDuration) ` Put this in a variable and then use it wherever required